### PR TITLE
`Development`: Ensure all endpoints return response entities

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Follow up for #8344 to enable the missing check and migrate our endpoints that violate the rule.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by Jan-Thurner — APPROVED** [review]
Code LGTM 👍

**Comment by b-fein** [comment]
I’d like to understand why for example returning a `ResponseEntity<OnlineResourceDTO>` is better than returning `OnlineResourceDTO` from a REST method. Can you please extend the PR description to explain/motivate why this change is useful?

**Comment by MaximilianAnzinger** [comment]
@b-fein
We discussed this and felt that consistency across all endpoints would be beneficial for the maintenance of this project. Most of our endpoints already apply this rule.
This way it's easier for new developers to distinguish between service methods and endpoints and it makes it clear that endpoints should not be used as service methods. (Yes, this is something we encountered in the past.)

**Review by Jan-Thurner — DISMISSED** [review]
reapprove

File: src/main/java/de/tum/in/www1/artemis/web/rest/ogparser/LinkPreviewResource.java [hunk 1]
```diff
@@ -32,7 +35,7 @@ public LinkPreviewResource(LinkPreviewService linkPreviewService) {
      * POST /link-preview : link preview for given url.
      *
      * @param url the url to parse
-     * @return the LinkPreviewDTO containing the meta information
+     * @return the ResponseEntity with status 200 (OK) and with body the LinkPreviewDTO containing the meta information
```
Comment (rstief) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ogparser/LinkPreviewResource.java:38]: We should revert the comment as we don't return a response entity here. (You said this class is exempt from returning a response entity, so this is fine)
```suggestion
     * @return the LinkPreviewDTO containing the meta information
```

**Review by rstief — APPROVED** [review]

**Review by rstief — APPROVED** [review]
code lgtm 👍

**Review by Jan-Thurner — APPROVED** [review]

**Pull request merged**